manager/manager.py

```diff
@@ -0,0 +1,1105 @@
+import copy
+import fnmatch
+import getpass
+import glob
+import os
+import stat
+import warnings
+from functools import wraps
+from pathlib import Path
+from types import SimpleNamespace
+from typing import Type, Union
+
+import appdirs
+import paramiko
+import yaml
+from ftpsync.ftp_target import FTPTarget
+from ftpsync.sftp_target import SFTPTarget
+from ftpsync.synchronizers import DownloadSynchronizer, UploadSynchronizer
+from ftpsync.targets import FsTarget
+
+# --------------------------------------------------------------------------------------------------------------------
+# Folder Class
+# --------------------------------------------------------------------------------------------------------------------
+
+
+class Folder:
+    """
+    Folder class used to contain details of canonical directories in the project directory tree.
+    """
+
+    def __init__(self, name, used, subfolders=None):
+        self.name = name
+        self.used = used
+        self.subfolders = subfolders
+
+
+# --------------------------------------------------------------------------------------------------------------------
+# Class Decorators
+# --------------------------------------------------------------------------------------------------------------------
+
+
+def requires_ssh_configs(func):
+    """
+    Decorator to check file is loaded. Used on Mainwindow class methods only as first arg is assumed to be self (containing cfgs)
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if not args[0].cfg["remote_host_id"] or not args[0].cfg["remote_host_username"]:
+            args[0]._raise_error(
+                "Cannot setup SSH connection, configuration file remote_host_id or remote_host_username is not set."
+            )
+        else:
+            return func(*args, **kwargs)
+
+    return wrapper
+
+
+# --------------------------------------------------------------------------------------------------------------------
+# Project Manager Class
+# --------------------------------------------------------------------------------------------------------------------
+
+
+class ProjectManager:
+    """
+    Main project manager class for data organisation and transfer in BIDS-style project directory.
+    The expected organisation is a central repository on a linux remote cluster ('remote') that
```

Comment:
Will this only ever work with a Linux remote? Just thinking of long-term generalisation.

Reply:
I think it should generalize easily, in fact if remote is mounted it does not require linux. The docstring is confusing because the only linux requirement is around writing the SSH key to authorized users, I will remove it and add it as a note on a TODO / issue